gcpdiag/lint/gke/err_2022_002_private_google_access.py

```diff
@@ -25,7 +25,7 @@ def run_rule(context: models.Context, report: lint.LintReportRuleInterface):
   if not clusters:
     report.add_skipped(None, 'no clusters found')
   for _, c in sorted(clusters.items()):
-    if c.is_private and not c.subnetwork.is_private_ip_google_access():
+    if c.is_private and not c.subnetwork.is_private_ip_google_access() and not c.is_shared_vpc:
```

Comment:
I guess I didn't think this through, but the issue might actually be with the project ID passed to the `get_router` function:
https://github.com/GoogleCloudPlatform/gcpdiag/blob/main/gcpdiag/lint/gke/err_2022_002_private_google_access.py#L30

Would it be the project passed to `--project`? If so, this will not return the correct router (considering it's configured on the shared vpc project)

Reply:
Yeah, definitely, it tries fetching the router from the wrong project. It should probably try both

Reply:
I suppose I can close this PR, unless the shared VPC property is desirable at all?

Reply:
@eyalzek thanks for bringing this problem. As @schweikert mentioned let's keep this simple. I propose to reuse existing solution based on: https://github.com/GoogleCloudPlatform/gcpdiag/blob/main/gcpdiag/lint/gke/warn_2022_003_firewall_rules_permission.py#L41-L46

and use proper project_id to get router for the shared vpc:
```
router = network.get_router(project_id=p, ...
```

Let us know if you need help on this.